tests/test_scripts.py

```diff
@@ -82,3 +84,123 @@ def test_proto_conversion(tmpdir):
+                        - string: "sites.ini"
+                        - number: 0xEDB88320
+                        - and:
+                          - or:
+                            - arch: i386
+                            - number: 4
+                            - not:
+                              - count(mnemonic(xor)): 5
+                          - not:
+                            - os: linux
+            """
+        ),
+        "rule_2": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 2
+                        scope: function
+                    features:
+                      - and:
+                        - string: "sites.ini"
+                        - arch: i386
+                        - basic block:
+                          - and:
+                            - api: setsockopt
+                            - count(mnemonic(mov)): 3
+            """
+        ),
+        "rule_3": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 3
+                        scope: function
+                    features:
+                      - or:
+                        - not:
+                          - os: linux
+                        - basic block:
+                          - and:
+                            - api: bind
+                            - count(mnemonic(mov)): 3
+            """
+        ),
+        "rule_4": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 4
+                        scope: function
+                    features:
+                      - not:
+                        - string: "expa"
+            """
+        ),
+    }
+
+    """
+        The rule_overlaps list represents the number of overlaps between each rule in the RULESET.
+        An overlap includes a rule overlap with itself.
+        The overlaps are like:
+        - Rule 0 has zero overlaps in RULESET
+        - Rule 1 overlaps with 3 other rules in RULESET
+        - Rule 4 overlaps with itself in RULESET
+        These overlap values indicate the number of rules with which
+        each rule in RULESET has overlapping features.
+    """
+    rule_overlaps = [0, 3, 4, 4, 1]
+
+    rule_dir = tmpdir.mkdir("capa_rule_overlap_test")
+    rule_paths = []
+
+    rule_file = tmpdir.join("%s.yml" % "rule_0")
+    rule_file.write(TEST_RULE_0)
+    rule_paths.append(rule_file.strpath)
+
+    for rule_name, RULE_CONTENT in TEST_RULESET.items():
+        rule_file = rule_dir.join("%s.yml" % rule_name)
+        rule_file.write(RULE_CONTENT)
+        rule_paths.append(rule_file.strpath)
+
+    # tests if number of overlaps for rules in RULESET found are correct.
+    for expected_overlaps, rule_path in zip(rule_overlaps, rule_paths):
+        overlaps_found = run_detect_duplicate_features(rule_dir.strpath, rule_path)
```

Comment:
we can call main from the script here directly

Reply:
I can go with both approaches, running it using pytest, or by passing the args to main from script.

Reply:
once the above is removed, we don't need the extra function anymore and can call the function to detect dups directly